### ISSUE TITLE
imdb lstm parameters explanation

### ISSUE THREAD

**Issue by vinayakumarr** [tflearn/tflearn#113]
Thanks,I am able to run imdb lstm example for my data set but accuracy is too low and i need to improve this. Once i am able to get good accuracy, I will try contributing to the project as well as soon as I finish the stuff that I am working on. I would like to know the below parameters or each line explanation with corresponding to my data set. I already mentioned my data set is 9 classes and around 1803 features. 
In imdb you used 

`maxlen=200` 
 `nb_classes=2 ` 
 `net = tflearn.input_data([None, 200])   `
 ` net = tflearn.embedding(net, input_dim=20000, output_dim=128)`
 `net = tflearn.lstm(net, 128) `
 `net = tflearn.dropout(net, 0.5)  `
 `net = tflearn.fully_connected(net, 2, activation='softmax')  `
 net = tflearn.regression(net, optimizer='adam', loss='categorical_crossentropy') //what this line exactly does
model = tflearn.DNN(net, clip_gradients=0., tensorboard_verbose=0) 
model.fit(trainX, trainY, validation_set=(testX, testY), show_metric=True,batch_size=128) 

- maxlen is standard length  What will be the size in my case?

- nb_classes is number of classes and in my case 10

- i understood from #8 that embedding is not required for my data set.is that right?

- input_dim=20000 dictionary size and in my case np.max(trainX)+1 but np.max(trainX)+1 is returning 2 as output is that correct and what will be output_dim for my dataset

- 2 number of classes and is there any other activation metods are availble. if who can you please specify

- batch_size=128 and what will be for my data set and i added one more extra parameters n_epoch=100 (because default is 10) but it is generating an error
